python/tvm/relay/quantize/quantize.py

```diff
@@ -0,0 +1,258 @@
+    """Configuration for quantization.
+
+    Parameters
+    ---------
+    nbit_dict: dict of QAnnotateKind -> int
+        Number of bit for every kind of annotate field.
+
+    global_scale: float
+        The global scale for calibration.
+
+    skip_k_conv: int
+        The number of skipped conv2d.
+    """
+    current = None
+
+    def __init__(self, nbit_dict, global_scale, skip_k_conv):
+        if nbit_dict is None:
+            nbit_dict = {
+                QAnnotateKind.INPUT: 8,
+                QAnnotateKind.WEIGHT: 8,
+                QAnnotateKind.ACTIVATION: 32
+            }
+        self.nbit_dict = nbit_dict
+        self.global_scale = global_scale
+        self.skip_k_conv = skip_k_conv
+        self.counter = 0
+
+    def __enter__(self):
+        self._old_scope = QConfig.current
+        QConfig.current = self
+        return self
+
+    def __exit__(self, ptype, value, trace):
+        assert self._old_scope
+        QConfig.current = self._old_scope
+
+
+QConfig.current = QConfig(None, 8.0, 1)
+
+
+def qconfig(nbit_dict=None, global_scale=8.0, skip_k_conv=1):
+    """Configuration for quantization.
+
+    Parameters
+    ---------
+    nbit_dict: dict of QAnnotateKind -> int
+        Number of bit for every kind of annotate field.
+
+    global_scale: float
+        The global scale for calibration.
+
+    skip_k_conv: int
+        The number of skipped conv2d.
+    """
+    return QConfig(nbit_dict, global_scale, skip_k_conv)
+
+
+def get_current_qconfig():
+    """Get the current quantization configuration."""
+    return QConfig.current
+
+
+@register_func("relay.quantize.get_config_bit")
+def get_config_bit(kind):
+    """Get the configured number of bit.
+
+    Parameters
+    ---------
+    kind: QAnnotateKind
+        the kind of annotation field.
+    """
+    cfg = get_current_qconfig()
+    return cfg.nbit_dict[kind]
+
+
+@register_relay_node
+class QAnnotateExpr(_expr.TempExpr):
+    """A special kind of Expr for Annotating.
+
+    Parameters
+    ---------
+    expr: Expr
+        the original relay ir expr.
+
+    kind: QAnnotateKind
+        the kind of annotation field.
+    """
+    def __init__(self, expr, kind):
+        self.__init_handle_by_constructor__(
+            _quantize.make_annotate_expr, expr, kind)
+
+
+@register_func("relay.quantize.attach_simulated_quantize")
+def attach_simulated_quantize(data, kind):
+    """Attach a simulated quantize operation after input data expr.
+
+    Parameters
+    ---------
+    data: Expr
+        the original data expr.
+
+    kind: QAnnotateKind
+        the kind of annotation field.
+    """
+    dom_scale = _expr.var("dom_scale")
+    bit = _expr.var("bit")
+    clip_min = _expr.var("clip_min")
+    clip_max = _expr.var("clip_max")
+    return _quantize.simulated_quantize(data, dom_scale, bit, clip_min,
+                                        clip_max, True, "round", kind)
+
+
+def register_annotate_function(op_name, frewrite=None, level=10):
```

Comment:
move this to _annotate.py

Reply:
I think this api can be exposed for rewriting annotating function?